homeworks/03-data-structures/src/binary-search-tree.ts

```diff
@@ -0,0 +1,46 @@
+import {BinaryTree, BinaryTreeImpl} from "./binary-tree"
+import {TraverseType, TreeNode} from "./tree-common"
+
+export interface BinarySearchTree extends BinaryTree<number> {
+    has(value: number): boolean
+}
+
+export class BinarySearchTreeImpl implements BinarySearchTree {
+    private binaryTree: BinaryTree<number>
+
+    constructor(tree: TreeNode<number>) {
+        this.binaryTree = new BinaryTreeImpl(tree)
+    }
+
+    root(): TreeNode<number> {
```

Comment:
This can be a public getter to shorten the syntax:

```typescript
export interface BinaryTree<T> {
    readonly root: TreeNode<T>
}

get root(): TreeNode<T> {
    return this._root
}
```

And everywhere in code you can just use `this.root` instead of `this.root()`

Reply:
Good point.